osx/KA-Lite/KA-Lite/AppDelegate.m

```diff
@@ -270,26 +269,34 @@ - (enum kaliteStatus)checkRunTask:(NSNotification *)aNotification{
     if (!isKaliteCommand(lastCommandArg)) {
         if (status == statusOkRunning) {
             self.portAlertCounter += 1;
-            [self getKaliteStatus];
         }
-        return self.status;
-    }
-    
-    if (self.kaliteProcessCounter >= 1) {
-        self.kaliteProcessCounter -= 1;
-    }
-    if (self.kaliteProcessCounter <= 0) {
+        [self getKaliteStatus];
         return self.status;
     }
     
     if (checkKaliteExecutable()) {
+        
         if ([lastCommandArg isEqualToString: @"status"]) {
-
+            
             //Check if KA Lite port 8008 is open.
             if ((self.portAlertCounter == 1) && (status != statusOkRunning)){
                 self.portAlertCounter -= 1;
                 alert(@"Port :8008 is occupied. Please close the process that is using it to start the KA Lite.");
+                [self setNewStatus:statusStopped];
                 [self showStatus:statusStopped];
+                return self.status;
+            } else {
+                if (self.portAlertCounter == 1) {
+                    self.portAlertCounter -= 1;
+                }
+                // This will run `kalite start` command after checking the port 8008.
+                if (self.kaliteStartCounter == 1) {
+                    self.kaliteStartCounter = statusOkRunning;
```

Comment:
Why did you assign an `enum` value here instead of an `int`?  Shouldn't this be an increment or decrement?  Basically this assigns a zero value to `kaliteStartCounter`, right?